### PR TITLE
doc: add documentation for workers module

### PR DESCRIPTION
**Pull request by addaleax**
##### Checklist

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/ayojs/ayo/blob/latest/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)

workers

### CODE REVIEW

File: doc/api/worker.md [hunk 1]
```diff
@@ -0,0 +1,465 @@
+# Worker
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental
+
+The `worker` module provides a way to create multiple environments running
+on independent threads, and to create message channels between those. It
```
Comment (ghost) [inline doc/api/worker.md]: `those` -> `them`

File: doc/api/worker.md [hunk 2]
```diff
@@ -0,0 +1,465 @@
...
+
+```js
+const worker = require('worker');
+```
+
+Workers are useful for performing CPU-intensive JavaScript operations; do not
+use them for I/O, since Ayo’s built-in mechanisms for performing operations
+asynchronously already treat it more efficiently than Worker threads can.
+
+Workers can also, unlike child processes or when using the `cluster` module,
```
Comment (ghost) [inline doc/api/worker.md]: move the whole between-commas section after `Workers`

File: doc/api/worker.md [hunk 3]
```diff
@@ -0,0 +1,465 @@
...
+An arbitrary JavaScript value that contains a clone of the data passed
+to this thread’s `Worker` constructor.
+
+
+## Class: MessageChannel
+<!-- YAML
+added: REPLACEME
+-->
+
+
```
Comment (ghost) [inline doc/api/worker.md]: unnecessary whitespace!

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by addaleax** [comment]
@pup addressed comments!

**Review by ghost — APPROVED** [review]

File: doc/api/worker.md [hunk 4]
```diff
@@ -0,0 +1,464 @@
...
+Send a message to the parent thread’s `Worker` instance that will be received
+via [`worker.on('message')`][]. See [`port.postMessage()`][] for
+more details.
+
+### worker.ref()
+<!-- YAML
+added: REPLACEME
+-->
+
+Opposite of `unref`, calling `ref` on a previously `unref`d worker will *not*
```
Comment (Qard) [inline doc/api/worker.md:95]: Feels a little weird to me referencing `unref` before it is explained. Maybe the `ref` and `unref` sections could swap places?

File: doc/api/worker.md [hunk 5]
```diff
@@ -0,0 +1,464 @@
...
+non-enumberable properties, accessors, and prototypes are not preserved.
+In particular, [`Buffer`][] objects will be read as plain [`Uint8Array`][]s
+on the receiving side.
+
+### port.ref()
+<!-- YAML
+added: REPLACEME
+-->
+
+Opposite of `unref`, calling `ref` on a previously `unref`d port will *not*
```
Comment (Qard) [inline doc/api/worker.md:244]: Again, maybe swap `ref` and `unref`, for clarity?

File: doc/api/worker.md [hunk 6]
```diff
@@ -0,0 +1,464 @@
...
+
+- The [`process.stdin`][], [`process.stdout`][] and [`process.stderr`][]
+  properties are set to `null`.
+- The [`domain`][] module is not usable inside of workers.
+- The [`require('worker').isMainThread`][] property is set to `false`.
+- The [`require('worker').postMessage()`][] method is available and the
+  [`require('worker').on('workerMessage')`][] event is being emitted.
+- [`process.exit()`][] does not stop the whole program, just the single thread,
+  and [`process.abort()`][] is not available.
+- [`process.chdir()`][] as well as `process` methods that set group or user ids
```
Comment (Qard) [inline doc/api/worker.md:288]: Should probably be more explicit and include the entire list here.

File: doc/api/worker.md (hunk 4 above)
Comment (addaleax) [reply doc/api/worker.md:95]: This is taken pretty much straight from `net.md` – and we do kind of have an “alphabetical” ordering rule, at least in Node…

Comment (Qard) [reply doc/api/worker.md:95]: Ah, yeah...I'm fine with it. 👍

**Comment by addaleax** [comment]
Landed in 717ca768beefb5a17d65eccfb68e4d520b5c4247
